server/test/unit/com/thoughtworks/go/remote/work/BuildWorkTest.java

```diff
@@ -629,4 +613,30 @@ private void createBuildWorkWithJobPlan(JobPlan jobPlan) throws Exception {
         );
         buildWork = new BuildWork(buildAssignment);
     }
+
+    @Test
+    public void shouldReportEnvironmentVariables() throws Exception {
+        buildWork = (BuildWork) getWork(WITH_ENV_VAR, PIPELINE_NAME);
+
+        buildWork.doWork(agentIdentifier, buildRepository, artifactManipulator, environmentVariableContext, AgentRuntimeInfo.fromAgent(agentIdentifier, "cookie", null), packageAsRepositoryExtension, scmExtension, taskExtension);
+
+        String consoleOut = artifactManipulator.consoleOut();
+
+        assertThat(trimTimeStamp(consoleOut), containsString("[go] setting environment variable 'GO_SERVER_URL' to value 'somewhere-does-not-matter'\n"
+                + "[go] setting environment variable 'GO_TRIGGER_USER' to value ''\n"
+                + "[go] setting environment variable 'GO_PIPELINE_NAME' to value 'pipeline1'\n"
```

Comment:
I wonder why we don't use:

```
export GO_PIPELINE_NAME='pipeline1'
```

I think it'll read better. And maybe:

```
export GO_PIPELINE_NAME='pipeline1' # Overridden
```

Easier to copy too (of course people will need to get rid of the "[go] " part.